binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/EventTypeRoutingWithInferredSerdeTests.java

```diff
@@ -0,0 +1,166 @@
+	@Test
+	void eventTypeRoutingWorksWhenInferredSerdeThrowsException() {
+		SpringApplication app = new SpringApplication(EventTypeRoutingWithConfiguredSerdeConfig.class);
+		app.setWebApplicationType(WebApplicationType.NONE);
+
+		try (ConfigurableApplicationContext context = app.run(
+			"--server.port=0",
+			"--spring.jmx.enabled=false",
+			"--spring.cloud.function.definition=process",
+			"--spring.cloud.stream.bindings.process-in-0.destination=foo-1",
+			"--spring.cloud.stream.bindings.process-out-0.destination=foo-2",
+			"--spring.cloud.stream.kafka.streams.bindings.process-in-0.consumer.eventTypes=foo,bar",
+			"--spring.cloud.stream.kafka.streams.bindings.process-in-0.consumer.useConfiguredSerdeWhenRoutingEvents=true",
+			"--spring.cloud.stream.kafka.streams.binder.deserializationExceptionHandler=logAndContinue",
+			"--spring.cloud.stream.kafka.streams.binder.functions.process.applicationId=process-id-foo-0",
+			"--spring.cloud.stream.kafka.streams.binder.configuration.commit.interval.ms=1000",
+			"--spring.cloud.stream.kafka.streams.binder.brokers=" + embeddedKafka.getBrokersAsString())) {
+			Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+			senderProps.put("value.serializer", IntegerSerializer.class);
+			DefaultKafkaProducerFactory<Integer, Integer> pf = new DefaultKafkaProducerFactory<>(senderProps);
+			try {
+				KafkaTemplate<Integer, Integer> template = new KafkaTemplate<>(pf, true);
+				template.setDefaultTopic("foo-1");
+
+				Headers headers = new RecordHeaders();
+				headers.add(new RecordHeader("event_type", "foo".getBytes()));
+
+				final ProducerRecord<Integer, Integer> producerRecord1 = new ProducerRecord<>("foo-1", 0, 56, 56, headers);
+				template.send(producerRecord1);
+
+
+				final ProducerRecord<Integer, Integer> producerRecord2 = new ProducerRecord<>("foo-1", 0, 57, 57);
+				template.send(producerRecord2);
+
+				final ProducerRecord<Integer, Integer> producerRecord3 = new ProducerRecord<>("foo-1", 0, 58, 58, headers);
+				template.send(producerRecord3);
+
+				Headers headers1 = new RecordHeaders();
+				headers1.add(new RecordHeader("event_type", "bar".getBytes()));
+
+				final ProducerRecord<Integer, Integer> producerRecord4 = new ProducerRecord<>("foo-1", 0, 59, 59, headers1);
+				template.send(producerRecord4);
+
+				senderProps.put("value.serializer", StringSerializer.class);
+				DefaultKafkaProducerFactory<Integer, String> pf1 = new DefaultKafkaProducerFactory<>(senderProps);
+				KafkaTemplate<Integer, String> template1 = new KafkaTemplate<>(pf1, true);
+				template1.setDefaultTopic("foo-1");
+				final ProducerRecord<Integer, String> producerRecordBar = new ProducerRecord<>("foo-1", 0, 67, "foobar", headers);
+				// This send should throw a deserializer exception by the Kafka Streams processor.
+				// Since the processor is provisioned with an exception handler of logAndContinue, it shouldn't affect anything.
+				template1.send(producerRecordBar);
```

Comment:
W/o the fix, what does this test do at this point? I am guessing its fine, but the consumer will barf when attempting to pull the invalid data from the routed topic?

Reply:
W/o the fix, the deserialization will fail from that branch ^^ and run into a runtime error. With the fix, apps can set the property to `true` and provide an exception handler (such as `logAndContinue` or `sendToDlq`).

Reply:
Basically, w/o the fix, the test, as it is, will fail at runtime.

Reply:
Got it, when that exception handler is not in play it would never reach the next line..